experimental/oak_baremetal_channel/src/server.rs

```diff
@@ -30,38 +30,38 @@ where
             inner: InvocationChannel::new(socket),
         }
     }
-    pub fn read_request(&mut self) -> anyhow::Result<Message> {
+    pub fn read_request(&mut self) -> anyhow::Result<message::RequestMessage> {
         let message = self.inner.read_message()?;
         Ok(message)
     }
-    pub fn write_response(&mut self, response: Message) -> anyhow::Result<()> {
+    pub fn write_response(&mut self, response: message::ResponseMessage) -> anyhow::Result<()> {
         self.inner.write_message(response)
     }
 }
 
-/// Construct a [`oak_idl::Request`] from a [`Message`].
-impl<'a> From<&'a Message> for oak_idl::Request<'a> {
-    fn from(frame: &'a Message) -> Self {
+/// Construct a [`oak_idl::Request`] from a [`message::RequestMessage`].
+impl<'a> From<&'a message::RequestMessage> for oak_idl::Request<'a> {
```

Comment:
I previously thought that `oak_idl::Request` was inside the `message::RequestMessage::data`, because it was a different layer in our protocol stack.